### PR TITLE
[MINOR][SQL][DOCS] Fix docs of Dataset.scala and SQLImplicits.scala.

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This PR fixes a sample code, a description, and indentations in docs.
## How was this patch tested?

Manual.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -93,7 +93,7 @@ private[sql] object Dataset {
  * to some files on storage systems, using the `read` function available on a `SparkSession`.
  * {{{
  *   val people = spark.read.parquet("...").as[Person]  // Scala
- *   Dataset<Person> people = spark.read().parquet("...").as(Encoders.bean(Person.class)  // Java
+ *   Dataset<Person> people = spark.read().parquet("...").as(Encoders.bean(Person.class))  // Java
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: Needs a semicolon at the end if this is Java?

**Comment by andrewor14** [comment]
LGTM I'll add the semicolon master 2.0

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 1 above)
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: Oh, nice catch. Right.

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen and @andrewor14 !

**Pull request closed without merging**